docs/fastdds/dds_layer/subscriber/dataReader/readingData.rst

```diff
@@ -0,0 +1,69 @@
+.. _dds_layer_subscriber_accessreceived:
+
+Accessing received data
+=======================
+
+The application can access and consume the data values received on the :ref:`dds_layer_subscriber_dataReader`
+by *reading* or *taking*.
+
+ * **Reading** is done with the :func:`read_next_data` member function.
+   It reads the next, non-previously accessed data value available on the
+   :ref:`dds_layer_subscriber_dataReader`, and stores it in the provided data buffer.
+ * **Taking** is done with the :func:`take_next_data` member function.
+   It reads the next, non-previously accessed data value available on the
+   :ref:`dds_layer_subscriber_dataReader`, and stores it in the provided data buffer.
+   Additionally, it also removes the value from the :ref:`dds_layer_subscriber_dataReader`,
+   so it is no longer accessible.
+
+If there is no unread data in the :ref:`dds_layer_subscriber_dataReader`, both operations will return
+``NO_DATA`` and nothing is returned.
+
+In addition to the data value, the data access operations also provide a :ref:`dds_layer_subscriber_sampleInfo`
+instance with additional information that help interpreting the returned data value, like the originating
+:ref:`dds_layer_publisher_dataWriter` or the publication time stamp.
+Please, refer to the :ref:`dds_layer_subscriber_sampleInfo` section for an extensive description of its contents.
+
+
+.. _dds_layer_subscriber_accessreceived_listener:
+
+Accessing data on callbacks
+---------------------------
+
+When the :ref:`dds_layer_subscriber_dataReader` new data values from any matching
+:ref:`dds_layer_publisher_dataWriter`, it informs the application through
+two Listener callbacks:
+
+* :cpp:func:`on_data_available<eprosima::fastdds::dds::DataReaderListener::on_data_available>`
```

Comment:
Can't this also be on the `Subscriber` or `DomainParticipant` Listeners? I understood that they inherit upwards, meaning that if the `DataReader` is created without listener, then the `SubscriberListener` is called, or else the `DomainParticipantListener`

Reply:
sure